utils/zerovec/src/map/vecs.rs

```diff
@@ -201,42 +200,34 @@ where
             .all(|w| T::from_unaligned(w[1]).cmp(&T::from_unaligned(w[0])) == Ordering::Greater)
     }
 
-    fn zvl_as_borrowed(&'a self) -> &'a ZeroSlice<T> {
+    fn zvl_as_borrowed(&self) -> &ZeroSlice<T> {
         &*self
     }
-    fn zvl_as_borrowed_inner(&self) -> Option<&'a ZeroSlice<T>> {
-        if let ZeroVec::Borrowed(b) = *self {
-            Some(ZeroSlice::from_ule_slice(b))
-        } else {
-            None
-        }
-    }
-
-    fn zvl_from_borrowed(b: &'a ZeroSlice<T>) -> Self {
-        b.as_zerovec()
-    }
 
     #[inline]
     fn zvl_get_as_t<R>(g: &Self::GetType, f: impl FnOnce(&T) -> R) -> R {
         f(&T::from_unaligned(*g))
     }
 }
 
-impl<'a, T> ZeroVecLike<'a, T> for &'a ZeroSlice<T>
+impl<T> ZeroVecLike<T> for ZeroSlice<T>
 where
     T: AsULE + Copy,
 {
     type GetType = T::ULE;
-    type BorrowedVariant = &'a ZeroSlice<T>;
+    type BorrowedVariant = ZeroSlice<T>;
 
-    fn zvl_new() -> Self {
-        ZeroSlice::from_ule_slice(&[])
+    fn zvl_new() -> Self
+    where
+        Self: Sized,
+    {
+        unreachable!()
```

Comment:
Is this the right way to handle this, where `Self: Sized` even though this type is clearly not sized?